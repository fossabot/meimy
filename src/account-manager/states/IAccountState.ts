type AuthType = 'password' | 'o365';
export interface IAccountState {
    authType: AuthType;
    eimToken: string[];
    id: string | null;
    lastConnect: Date | null;
    siteDomain: string;
    siteName: string;
    userId?: string;
    password?: string;
}

export const createInitAccountState = (): IAccountState => {
    return {
        authType: 'o365',
        eimToken: [],
        id: null,
        lastConnect: null,
        siteDomain: '',
        siteName: '',
    };
};
