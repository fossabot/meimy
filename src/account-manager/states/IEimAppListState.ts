export interface IEimApp {
    appKey: string;
    appName: string;
    description: string;
    siteDomain: string;
    siteName: string;
    tokens: string[];
}

export default interface IEimAppList {
    appList: IEimApp[];
    loading: boolean;
};;;;;;;;;;

export const createInitEimAppList = (): IEimAppList => {
    return {
        appList: [],
        loading: false,
    };
};
