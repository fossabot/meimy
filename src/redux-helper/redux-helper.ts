import Clone from 'clone';
import { NavigationContainer, NavigationScreenProp, NavigationState } from 'react-navigation';
import {
    createNavigationReducer,
    createReactNavigationReduxMiddleware,
    ReducerState,
} from 'react-navigation-redux-helpers';
import { Action, AnyAction, applyMiddleware, combineReducers, createStore, Dispatch, ReducersMapObject } from 'redux';

export interface IProps<S> {
    state: S;
}

interface INavState {
    key: string;
    params: object;
    routeName: string;
}

interface INavProps {
    navigation: NavigationScreenProp<INavState>;
    dispatch: Dispatch;
}

export interface ICombinedNavProps<S>
    extends IProps<S>,
    INavProps { }

export interface IStateAtNavigation {
    nav: ReducerState;
}

// eslint-disable-next-line @typescript-eslint/no-explicit-any
type WorkOfAction<S, A extends Action> = (
    state: S,
    action: A,
) => S | void;

/***
* action に対する reducer の処理を管理する
*/
class ActionToReducerMapper<S> {
    /** アクション・タイプと処理の定義を保持する。 */
    private works: { [actionKey: string]: WorkOfAction<S, AnyAction> } = {};
    /** アクション・タイプと処理の定義を追加する。 */
    public addWork = <A extends AnyAction>(
        actionType: string,
        func: WorkOfAction<S, A>,
    ): void => {
        this.works[actionType] = func as WorkOfAction<S, AnyAction>;
    }
    /*** 処理を実行する。
    * 該当するアクション・タイプがあった場合、state をクローンして指定の処理を行い、返す。
    * 該当するアクション・タイプが無い場合、何も処理を行わず、state もクローンせずにそのまま返す。
    */
    public execute = (state: S, action: Action) => {
        let newState = state;
        const process = this.works[action.type];
        if (!!process) {
            newState = Clone(state);
            const retState = process(newState, action);
            if (!!retState) {
                newState = retState;
            }
        }
        return newState;
    }
}

export const createActionToReducerMapper = <S>() => {
    return new ActionToReducerMapper<S>();
};

export const generateStore = <S>(
    navContainer: NavigationContainer,
    userMapper: ReducersMapObject<S>,
) => {
    type IMixState = S & IStateAtNavigation;
    const navReducer = createNavigationReducer(navContainer);
    const mixReducers = Object.assign(
        { nav: navReducer },
        userMapper,
    );
    const reducer = combineReducers(mixReducers);

    const middleware = createReactNavigationReduxMiddleware(
        (state: S & IStateAtNavigation) => state.nav as NavigationState,
    );
    const store = createStore(reducer, applyMiddleware(middleware));
    if ('' === '') {

    }
    return store;
};
