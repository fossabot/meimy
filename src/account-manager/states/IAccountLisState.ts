import { IAccountState } from './IAccountState';
import { IAuthState } from './IAuthStates';

export interface IAccountListState {
    accounts: IAccountState[];
    authState: IAuthState;
}

export const createInitAccountListState = (): IAccountListState => {
    return {
        accounts: [],
        authState: {},
    };
};
