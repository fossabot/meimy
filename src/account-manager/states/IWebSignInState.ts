import { IAccountState } from './IAccountState';

export default interface IWebSignInState {
    account?: IAccountState;
}
export const createInitWebSignInState = (): IWebSignInState => {
    return {
        account: undefined,
    };
};
