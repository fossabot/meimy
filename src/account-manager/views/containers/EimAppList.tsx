import { Body, Container, ListItem, Spinner, Text } from 'native-base';
import React, { Component } from 'react';
import { BackHandler } from 'react-native';
import { connect } from 'react-redux';

import { ICombinedNavProps, IProps } from '../../../redux-helper/redux-helper';
import { createLoadAppListAction } from '../../actions/EimAppListActions';
import NavigateActions from '../../actions/NavigateActions';
import { getConfig } from '../../Config';
import { IAccountManagerState } from '../../IAccountManagerState';
import IEimAppList from '../../states/IEimAppListState';

const config = getConfig();
class EimAppList extends Component<ICombinedNavProps<IEimAppList>> {
    public static navigationOptions = () => {
        return {
            headerStyle: {
                backgroundColor: config.colorPalets.$colorPrimary3,
            },
            headerTintColor: config.colorPalets.$invertColor,
            headerTitle: 'アプリ一覧',
        };
    }
    public constructor(props: ICombinedNavProps<IEimAppList>) {
        super(props);
        BackHandler.addEventListener('hardwareBackPress', () => {
            this.props.navigation.pop();
            return true;
        });
    }
    public render = () => {
        const appListBox = this.props.state.appList.map((app) => {
            return (
                <ListItem noIndent key={app.appKey} onPress={this.onPressItem.bind(this, app.appKey)}>
                    <Body>
                        <Text>{app.appName}</Text>
                        <Text note>{app.description}</Text>
                    </Body>
                </ListItem>
            );
        });
        return (
            <Container>
                {this.props.state.loading ? <Spinner color="blue" /> : appListBox}
            </Container>
        );
    }
    public componentDidMount = () => {
        createLoadAppListAction(this.props.dispatch, NavigateActions);
    }
    public onPressItem = (appKey: string) => {
        NavigateActions.openApp({ appKey }, this.props.navigation);
    }
}

const mapStateToProps = (state: IAccountManagerState): IProps<IEimAppList> => {
    return {
        state: state.appList,
    };
};

export default connect(mapStateToProps)(EimAppList);
