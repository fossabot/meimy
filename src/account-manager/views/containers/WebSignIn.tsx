import { Container, Toast } from 'native-base';
import React, { Component } from 'react';
import { NativeSyntheticEvent, NavState, WebView, WebViewMessageEventData, WebViewUriSource } from 'react-native';
import CookieManager from 'react-native-cookies';
import { connect } from 'react-redux';
import UrlParse from 'url-parse';

import { ICombinedNavProps, IProps } from '../../../redux-helper/redux-helper';
import { asyncSaveAccountAction } from '../../actions/AccountActions';
import NavigateActions from '../../actions/NavigateActions';
import config from '../../Config';
import { IAccountManagerState } from '../../IAccountManagerState';
import { IAccountListState } from '../../states/IAccountLisState';
import IWebSignInState from '../../states/IWebSignInState';

const ricohSamlUrl = 'https://adfs.jp.ricoh.com/adfs/ls/';

interface IWebSignInProps extends IWebSignInState {
    accountListState: IAccountListState;
}
type ThisProps = ICombinedNavProps<IWebSignInProps>;
interface ILocalState {
    uriSource: WebViewUriSource;
}
interface IWebViewMessage {
    userId: string;
    password: string;
}


// 画面内にログインフォームを表示する
const eimLoginFormSet = `
(function(){
    var eimSys = document.getElementById("eim-system");
    if (!!eimSys) {
        eimSys.style.minWidth = "0";
    }
})();`;
// EIMフォーム ユーザーパスワードの取得
const getEimUserIdPass = `
(function(){
    var form = document.querySelector("[name=form1]");
    if (!form) { return; }
    form.addEventListener("submit", function() {
        var userNameInput = document.querySelector("[name=userName]");
        var passwordInput = document.querySelector("[name=password]");
        var userPass = {userId: '', password: ''};
        if (!!userNameInput) { userPass.userId = userNameInput.value; }
        if (!!passwordInput) { userPass.password = passwordInput.value; }
        window.postMessage(JSON.stringify(userPass));
    });
})();`;

// 365 SAML フォーム ユーザーパスワードの取得
const get365UserIdPass = `
(function(){
    var userNameInput = document.querySelector("#userNameInput");
    var passwordInput = document.querySelector("#passwordInput");
    var button = document.querySelector("#submitButton");
    if (!button || !userNameInput || !passwordInput) { return; }
    button.addEventListener("click", function() {
        var userPass = {userId: '', password: ''};
        userPass.userId = userNameInput.value;
        userPass.password = passwordInput.value;
        window.postMessage(JSON.stringify(userPass));
    });
})();`;

class WebSignIn extends Component<ThisProps, ILocalState> {
    public static navigationOptions = () => {
        return {
            headerStyle: {
                backgroundColor: config.colorPalets.$colorPrimary3,
            },
            headerTintColor: config.colorPalets.$invertColor,
            headerTitle: '',
        };
    }
    private postedIdPass = false;
    private webview: WebView | null = null;
    private saveProp: ICombinedNavProps<IWebSignInProps> | undefined;
    public constructor(props: ThisProps) {
        super(props);
        const { account } = this.props.state;
        const domain = account ? account.siteDomain : '';
        let eimUrl = 'https://' + domain;
        if (!!account && account.authType === 'password') {
            eimUrl = eimUrl + '/services/login';
        }
        this.state = {
            uriSource: { uri: eimUrl },
        };
    }
    public render() {
        const { account } = this.props.state;
        if (!account) { return null; }
        const script = `
${account.authType === 'o365' ? '' : eimLoginFormSet}
${account.authType === 'o365' ? get365UserIdPass : getEimUserIdPass}
`;
        return (
            <Container>
                <WebView
                    source={this.state.uriSource}
                    onLoadStart={this.onLoadStartWebView}
                    onMessage={this.onMessage}
                    ref={(control) => { this.webview = control; }}
                    injectedJavaScript={script} />
            </Container>
        );
    }
    public onMessage = (event: NativeSyntheticEvent<WebViewMessageEventData>) => {
        const { data: message } = event.nativeEvent;
        const messageData = JSON.parse(message) as IWebViewMessage;
        /** ID/Password の保存 */
        const account = this.props.state.account;
        if (!account) { return; }
        account.userId = messageData.userId;
        account.password = messageData.password;
        asyncSaveAccountAction(account, this.props.dispatch);
    }
    public componentDidMount = () => {
        CookieManager.clearAll();
        // onLoadStartWebViewでは、state　がなくなる・・・？のでクラスメンバーとして保持する
        this.saveProp = this.props;
    }
    private postIdPassEimForm = (url: string) => {
        const { account } = this.props.state;
        if (!account) { return; }
        // services/v1/login
        if (this.postedIdPass) { return false; }
        if (!url.endsWith('/services/login')) { return false; }
        if (!account.userId || !account.password) {
            // 空の場合は初回処理の場合なので、POST処理しない
            this.postedIdPass = true;
            return false;
        }
        const body = `userName=${encodeURIComponent(account.userId)}`
            + `&password=${encodeURIComponent(account.password)}`;
        const postUrl = UrlParse(url);
        postUrl.set('pathname', '/services/v1/login');
        const uriSource: WebViewUriSource = {
            body,
            headers: {
                'Content-Type': 'application/x-www-form-urlencoded;charset=UTF-8',
            },
            method: 'POST',
            uri: postUrl.href,
        };
        this.postedIdPass = true;
        this.setState({
            uriSource,
        });
        return true;
    }
    private postIdPassFor365 = (url: string): boolean => {
        const account = this.props.state.account;
        if (!account) { return false; }
        if (this.postedIdPass) { return false; }
        if (!url.startsWith(ricohSamlUrl)) { return false; }
        if (!account.userId || !account.password) {
            // 空の場合は初回処理の場合なので、POST処理しない
            this.postedIdPass = true;
            return false;
        }
        const body = `AuthMethod=${encodeURIComponent('FormsAuthentication')}`
            + `&UserName=${encodeURIComponent(account.userId)}`
            + `&Password=${encodeURIComponent(account.password)}`;
        const uriSource: WebViewUriSource = {
            body,
            headers: {
                'Content-Type': 'application/x-www-form-urlencoded;charset=UTF-8',
            },
            method: 'POST',
            uri: url,
        };
        this.postedIdPass = true;
        this.setState({
            uriSource,
        });
        return true;
    }
    private onLoadStartWebView = (e: NavState) => {
        const { saveProp } = this;
        const account = saveProp ? saveProp.state.account : null;
        if (!account) { return; }
        // 引数の型が、 @types と異なるのでキャストし直す
        const navState = (e as NativeSyntheticEvent<NavState>).nativeEvent;
        const { url } = navState
        if (!url) { return; }
        const webview = this.webview;
        if (!webview) { return; }
        if (account.authType === 'o365' && this.postIdPassFor365(url)) {
            webview.stopLoading();
            return;
        }
        if (account.authType === 'password' && this.postIdPassEimForm(url)) {
            webview.stopLoading();
            return;
        }
        (async () => {
            // console.log(JSON.stringify(navState.nativeEvent));
            const cookie = await CookieManager.get(url);
            if (!cookie.APISID) { return; }
            // トークンのクッキーが取得できたら成功
            webview.stopLoading();
            Toast.show({ text: '認証に成功しました。', type: 'success' });
            account.eimToken = ['APISID=' + cookie.APISID];
            asyncSaveAccountAction(account, this.props.dispatch);
            if (!this.saveProp) { return; }

            NavigateActions.navigateForLink(
                this.props.state.accountListState,
                {
                    siteDomain: account.siteDomain,
                    siteName: account.siteName,
                    tokens: account.eimToken,
                }, this.saveProp.dispatch, this.saveProp.navigation, true);
        })();
    }
}

const mapStateToProps = (state: IAccountManagerState): IProps<IWebSignInProps> => {
    return {
        state: { ...state.webSignIn, accountListState: state.accountList },
    };
};

export default connect(mapStateToProps)(WebSignIn);
