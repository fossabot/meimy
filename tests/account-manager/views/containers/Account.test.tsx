import clone from 'clone';
import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import toJson from 'enzyme-to-json';
import React from 'react';

import { AuthType } from '../../../../src/account-manager/states/IAccountState';
import { _Account, IAccountProps, ILocaleState } from '../../../../src/account-manager/views/containers/Account';
import { ICombinedNavProps } from '../../../../src/redux-helper/redux-helper';

Enzyme.configure({ adapter: new Adapter() });

const getProps = (): ICombinedNavProps<IAccountProps> => {
    return {
        navigation: {
            setParams: jest.fn(),
            pop: jest.fn(),
        } as any,
        dispatch: jest.fn(),
        state: {
            authType: 'password' as AuthType,
            eimToken: [],
            id: '',
            lastConnect: null,
            siteDomain: '',
            siteName: '',
            accountListState: {
                accounts: [],
                authState: {},
            },
        }
    };
};
let props: ICombinedNavProps<IAccountProps>;
describe('init', () => {
    beforeEach(() => {
        props = getProps();
    });
    test('new account', () => {
        const wrapper = Enzyme.shallow(<_Account {...props} />);
        expect(toJson(wrapper)).toMatchSnapshot();
        expect(wrapper.contains('サイトドメインが正しくありません')).toBeTruthy();
        expect(wrapper.contains('サイト名称は必須です')).toBeTruthy();
        expect(wrapper.contains('サインイン')).toBeFalsy();
        expect(wrapper.exists('FloatingButton[iconName="md-checkmark"]')).toBeTruthy();
        expect(wrapper.exists('FloatingButton[iconName="close"]')).toBeTruthy();
        expect(wrapper.exists('FloatingButton[iconName="create"]')).toBeFalsy();
    });
    test('exist account', () => {
        props.state.id = '001';
        props.state.siteDomain = 'domain.eim.ricoh.com';
        props.state.siteName = 'name';
        const wrapper = Enzyme.shallow(<_Account {...props} />);
        expect(toJson(wrapper)).toMatchSnapshot();
        expect(wrapper.contains('サイトドメインが正しくありません')).toBeFalsy();
        expect(wrapper.contains('サイト名称は必須です')).toBeFalsy();
        expect(wrapper.contains('サインイン')).toBeTruthy();
        expect(wrapper.exists('FloatingButton[iconName="md-checkmark"]')).toBeFalsy();
        expect(wrapper.exists('FloatingButton[iconName="close"]')).toBeFalsy();
        expect(wrapper.exists('FloatingButton[iconName="create"]')).toBeTruthy();
    });
    test('exist account, sing in', () => {
        props.state.id = '001';
        props.state.siteDomain = 'domain.eim.ricoh.com';
        props.state.siteName = 'name';
        const wrapper = Enzyme.shallow(<_Account {...props} />);
        wrapper.setState({ loginProcessing: true });
        expect(toJson(wrapper)).toMatchSnapshot();
        expect(wrapper.contains('サインイン')).toBeFalsy();
        const spinner = wrapper.findWhere(n => n.key() === 'sigin_in_spinner');
        expect(spinner.name()).toEqual('Styled(Spinner)');
    });
});

describe('events', () => {
    beforeEach(() => {
        props = getProps();
    });

    test('true change', () => {
        props.state.id = '001';
        props.state.siteDomain = 'domain.eim.ricoh.com';
        props.state.siteName = 'name';
        props.state.authType = 'password';

        const wrapper = Enzyme.shallow(<_Account {...props} />);
        // 変更前
        expect((wrapper.state() as any)['siteName']).toEqual('name');
        expect((wrapper.state() as any)['siteDomain']).toEqual('domain.eim.ricoh.com');
        expect((wrapper.state() as any)['authType']).toEqual('password');
        const cloneState = clone(wrapper.state()) as ILocaleState;
        // 変更後
        const siteNameInput = wrapper.findWhere(n => {
            return n.key() === 'site_name';
        });
        siteNameInput.simulate('changeText', 'change-name');
        cloneState.siteName = 'change-name';
        expect(wrapper.state()).toEqual(cloneState);

        const siteDomainInput = wrapper.findWhere(n => {
            return n.key() === 'site_domain';
        });
        siteDomainInput.simulate('changeText', 'domain1.eim.ricoh.com');
        cloneState.siteDomain = 'domain1.eim.ricoh.com';
        expect(wrapper.state()).toEqual(cloneState);

        const authTypePicker = wrapper.findWhere(n => {
            return n.key() === 'auth-type';
        });
        authTypePicker.simulate('valueChange', 'o365');
        cloneState.authType = 'o365';
        expect(wrapper.state()).toEqual(cloneState);
    });
    test('on press edit button', () => {
        props.state.id = '001';
        props.state.siteDomain = 'domain.eim.ricoh.com';
        props.state.siteName = 'name';
        const wrapper = Enzyme.shallow(<_Account {...props} />);
        const button = wrapper.find('FloatingButton[iconName="create"]');
        expect((wrapper.instance() as _Account)['backupState']).toBeNull();
        const beforeState = wrapper.state();
        button.simulate('press');
        expect((wrapper.instance() as _Account)['backupState']).toStrictEqual(beforeState);
        expect((wrapper.state() as any).mode).toEqual('edit');
        expect(wrapper.exists('FloatingButton[iconName="md-checkmark"]')).toBeTruthy();
        expect(wrapper.exists('FloatingButton[iconName="close"]')).toBeTruthy();
        expect(wrapper.exists('FloatingButton[iconName="create"]')).toBeFalsy();
    });
});
