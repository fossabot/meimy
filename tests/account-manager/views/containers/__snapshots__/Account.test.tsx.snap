// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`init exist account 1`] = `
<Styled(Container)>
  <Styled(Content)>
    <Styled(Card)
      style={
        Object {
          "paddingBottom": 24,
        }
      }
    >
      <Styled(Item)
        stackedLabel={true}
      >
        <Styled(Label)>
          サイト名称
        </Styled(Label)>
        <Styled(Input)
          editable={false}
          key="site_name"
          onChangeText={[Function]}
          style={
            Object {
              "color": "#999",
            }
          }
          value="name"
        />
      </Styled(Item)>
      <Styled(Text)
        style={
          Object {
            "color": undefined,
            "fontSize": 12,
          }
        }
      />
      <Styled(Item)
        stackedLabel={true}
      >
        <Styled(Label)>
          サイトドメイン
        </Styled(Label)>
        <Styled(Input)
          autoCapitalize="none"
          editable={false}
          key="site_domain"
          keyboardType="url"
          onChangeText={[Function]}
          style={
            Object {
              "color": "#999",
            }
          }
          value="domain.eim.ricoh.com"
        />
      </Styled(Item)>
      <Styled(Text)
        style={
          Object {
            "color": undefined,
            "fontSize": 12,
          }
        }
      />
      <Styled(Item)
        style={
          Object {
            "marginBottom": 12,
            "marginTop": 12,
          }
        }
      >
        <Styled(Label)>
          認証方式
        </Styled(Label)>
        <Styled(PickerNB)
          enabled={false}
          iosIcon={
            <Styled(Icon)
              name="ios-arrow-down"
            />
          }
          key="auth-type"
          mode="dropdown"
          onValueChange={[Function]}
          selectedValue="password"
          style={Object {}}
          textStyle={
            Object {
              "color": "#999",
            }
          }
        >
          <Item
            label="パスワード"
            value="password"
          />
          <Item
            label="Office365認証"
            value="o365"
          />
        </Styled(PickerNB)>
      </Styled(Item)>
      <Styled(Button)
        block={true}
        disabled={false}
        onPress={[Function]}
        rounded={true}
        style={
          Object {
            "marginTop": 24,
          }
        }
        success={true}
      >
        <Styled(Text)>
          サインイン
        </Styled(Text)>
      </Styled(Button)>
    </Styled(Card)>
  </Styled(Content)>
  <FloatingButton
    iconColor="white"
    iconName="create"
    iconType="Ionicons"
    key="edit_button"
    label=""
    marginHorizontal={32}
    marginVertical={32}
    onPress={[Function]}
    position="bottom-right"
    primaryButtonColor="#000099"
    primaryButtonSize={56}
    secondaryButtonSize={48}
  />
</Styled(Container)>
`;

exports[`init exist account, sing in 1`] = `
<Styled(Container)>
  <Styled(Content)>
    <Styled(Card)
      style={
        Object {
          "paddingBottom": 24,
        }
      }
    >
      <Styled(Item)
        stackedLabel={true}
      >
        <Styled(Label)>
          サイト名称
        </Styled(Label)>
        <Styled(Input)
          editable={false}
          key="site_name"
          onChangeText={[Function]}
          style={
            Object {
              "color": "#999",
            }
          }
          value="name"
        />
      </Styled(Item)>
      <Styled(Text)
        style={
          Object {
            "color": undefined,
            "fontSize": 12,
          }
        }
      />
      <Styled(Item)
        stackedLabel={true}
      >
        <Styled(Label)>
          サイトドメイン
        </Styled(Label)>
        <Styled(Input)
          autoCapitalize="none"
          editable={false}
          key="site_domain"
          keyboardType="url"
          onChangeText={[Function]}
          style={
            Object {
              "color": "#999",
            }
          }
          value="domain.eim.ricoh.com"
        />
      </Styled(Item)>
      <Styled(Text)
        style={
          Object {
            "color": undefined,
            "fontSize": 12,
          }
        }
      />
      <Styled(Item)
        style={
          Object {
            "marginBottom": 12,
            "marginTop": 12,
          }
        }
      >
        <Styled(Label)>
          認証方式
        </Styled(Label)>
        <Styled(PickerNB)
          enabled={false}
          iosIcon={
            <Styled(Icon)
              name="ios-arrow-down"
            />
          }
          key="auth-type"
          mode="dropdown"
          onValueChange={[Function]}
          selectedValue="password"
          style={Object {}}
          textStyle={
            Object {
              "color": "#999",
            }
          }
        >
          <Item
            label="パスワード"
            value="password"
          />
          <Item
            label="Office365認証"
            value="o365"
          />
        </Styled(PickerNB)>
      </Styled(Item)>
      <Styled(Spinner)
        color="#2F4172"
        key="sigin_in_spinner"
      />
    </Styled(Card)>
  </Styled(Content)>
  <FloatingButton
    iconColor="white"
    iconName="create"
    iconType="Ionicons"
    key="edit_button"
    label=""
    marginHorizontal={32}
    marginVertical={32}
    onPress={[Function]}
    position="bottom-right"
    primaryButtonColor="#000099"
    primaryButtonSize={56}
    secondaryButtonSize={48}
  />
</Styled(Container)>
`;

exports[`init new account 1`] = `
<Styled(Container)>
  <Styled(Content)>
    <Styled(Card)
      style={
        Object {
          "paddingBottom": 24,
        }
      }
    >
      <Styled(Item)
        stackedLabel={true}
      >
        <Styled(Label)>
          サイト名称
        </Styled(Label)>
        <Styled(Input)
          editable={true}
          key="site_name"
          onChangeText={[Function]}
          style={
            Object {
              "color": "#333",
            }
          }
          value=""
        />
      </Styled(Item)>
      <Styled(Text)
        style={
          Object {
            "color": undefined,
            "fontSize": 12,
          }
        }
      >
        サイト名称は必須です
      </Styled(Text)>
      <Styled(Item)
        stackedLabel={true}
      >
        <Styled(Label)>
          サイトドメイン
        </Styled(Label)>
        <Styled(Input)
          autoCapitalize="none"
          editable={true}
          key="site_domain"
          keyboardType="url"
          onChangeText={[Function]}
          style={
            Object {
              "color": "#333",
            }
          }
          value=""
        />
      </Styled(Item)>
      <Styled(Text)
        style={
          Object {
            "color": undefined,
            "fontSize": 12,
          }
        }
      >
        サイトドメインが正しくありません
      </Styled(Text)>
      <Styled(Item)
        style={
          Object {
            "marginBottom": 12,
            "marginTop": 12,
          }
        }
      >
        <Styled(Label)>
          認証方式
        </Styled(Label)>
        <Styled(PickerNB)
          enabled={true}
          iosIcon={
            <Styled(Icon)
              name="ios-arrow-down"
            />
          }
          key="auth-type"
          mode="dropdown"
          onValueChange={[Function]}
          selectedValue="password"
          style={Object {}}
          textStyle={
            Object {
              "color": "#333",
            }
          }
        >
          <Item
            label="パスワード"
            value="password"
          />
          <Item
            label="Office365認証"
            value="o365"
          />
        </Styled(PickerNB)>
      </Styled(Item)>
    </Styled(Card)>
  </Styled(Content)>
  <FloatingButton
    iconColor="white"
    iconName="md-checkmark"
    iconType="Ionicons"
    key="save_button"
    label=""
    marginHorizontal={32}
    marginVertical={32}
    onPress={[Function]}
    position="bottom-right"
    primaryButtonColor="#000099"
    primaryButtonSize={56}
    secondaryButtonSize={48}
  />
  <FloatingButton
    iconColor="white"
    iconName="close"
    iconType="Ionicons"
    key="cancel"
    label=""
    marginHorizontal={32}
    marginVertical={32}
    onPress={[Function]}
    position="bottom-left"
    primaryButtonColor="#000099"
    primaryButtonSize={56}
    secondaryButtonSize={48}
  />
</Styled(Container)>
`;
